### PR TITLE
[#92466170] Handle invalid data from Forte

### PR DESCRIPTION
**Pull request by elijahr**
Story: https://www.pivotaltracker.com/story/show/92466170

Assigned @dlinder

### CODE REVIEW

**Comment by dlinder** [comment]
Thinking back to writing this originally, I went out of my way to make sure the PaymentForm would always validate and therefore save. Good call though logging if that's not happening.

:fire: 
:tropical_drink: :moyai:

File: satchless/payment/__init__.py [hunk 1]
```diff
@@ -4,6 +4,14 @@ def __init__(self, error_message):
         self.error_message = error_message
 
 
+class PaymentsGatewayReceiptFormError(PaymentFailure):
+    def __init__(self, form):
+        error_message = ', '.join(
```
Comment (jgaren) [inline satchless/payment/__init__.py]: What about non-field (global) errors?

Comment (elijahr) [inline satchless/payment/__init__.py]: Those are in `form.errors['__all__']` so they would be displayed.

Comment (jgaren) [inline satchless/payment/__init__.py]: I believe you are thinking of `form._errors`, which is distinct from form.errors.

Comment (elijahr) [inline satchless/payment/__init__.py]: Ah! Yes.  Changed in 0f684ee05a0b7eb17a1e96a6c030fb7e14969349

Comment (jgaren) [inline satchless/payment/__init__.py]: Think you may want u'' here for the separator, errors tend to be unicode.

Could also switch from `items` to `iteritems` for efficiency.

Comment (elijahr) [inline satchless/payment/__init__.py]: Fixed in 8688eb0075f069756623fb12481ae300c591bccf.  Good suggestions - thank you kindly.

**Pull request merged**